File: solutions/patterns/star-patterns/09-hourglass-pattern/solution.java
import java.util.*;

class solution {

    public static void main(String args[]) {
        int n = 4;

        for (int i = 1; i <= 2 * n - 1; i++) {

            // Top inverted pyramid
            if (i <= n) {
                // spaces
                for (int j = 1; j < i; j++) {
                    System.out.print(" ");
                }
                // stars
                for (int j = i; j <= n; j++) {
                    System.out.print("* ");
                }
            } 
            // Bottom pyramid
            else {
                // spaces
                for (int j = 1; j <= 2 * n - i - 1; j++) {
                    System.out.print(" ");
                }
                // stars
                for (int j = n; j >= 2 * n - i; j--) {
                    System.out.print("* ");
                }
            }

            System.out.println();
        }
    }
}

File: solutions/patterns/star-patterns/09-hourglass-pattern/solution.js
let n = 5; // height of pyramid

for (let i = 0; i < n; i++) {
    let row = "";

    // spaces
    for (let j = 0; j <=i; j++) {
        row += " ";
    }

    // stars
    for (let k = 0; k <n-i; k++) {
        row += "* ";
    }

    // print the row
    console.log(row);
}

for (let i = 0; i < n; i++) {
    let row = "";

    // spaces
    for (let j = 0; j < n - i - 1; j++) {
        row += " ";
    }

    // stars
    for (let k = 0; k <=i; k++) {
        row += " *";
    }

    // print the row
    console.log(row);
}


//By The Architect 🙎‍♂️